tests/php/ImageManipulationTest.php

```diff
@@ -427,4 +427,28 @@ public function testRender(string $template, string $expected)
             trim($image->renderWith(SSViewer::fromString($template)))
         );
     }
+
+    public function testThumbnailURL()
+    {
+        $img = $this->objFromFixture(Image::class, 'imageWithTitle');
+
+        // File needs to be in draft and users need to be anonymous to test the access
+        $this->logOut();
+        $img->doUnpublish();
+
+        $fileUrl = 'folder/444065542b/test-image__FillWzEwLDEwXQ.png';
+
+        $this->assertEquals(
+            '/assets/' . $fileUrl,
+            $img->ThumbnailURL(10, 10),
+            'Thumbnail URL is correct'
+        );
+
+        /** @var AssetStore assetStore */
+        $assetStore = Injector::inst()->get(AssetStore::class);
+        $this->assertFalse(
+            $assetStore->isGranted($fileUrl),
+            'Current user is not automatically granted access'
```

Comment:
```suggestion
            'Logged out user was granted access to draft resampled image'
```

Reply:
I prefer assertion message to describe what is being asserted, rather than what has failed. 

But I can see the message here could be confusing if you think `Current user is not automatically granted access` describes what has failed.

What about this instead?
 ```suggestion
            'Current user should not automatically be granted access to view thumbnail'
```